File: app/src/main/java/com/wbrawner/simplemarkdown/model/Sentence.java
package com.wbrawner.simplemarkdown.model;

import eu.crydee.syllablecounter.SyllableCounter;

public class Sentence {

    private String sentence = "";
    private int start = 0;
    private int end = 0;

    private final static SyllableCounter sc = new SyllableCounter();

    public Sentence(String content, int start, int end){
        this.start = start;
        this.end = end;
        this.sentence = content.substring(start, end);

        trimStart();
    }

    public Sentence(String sentence){
        this.sentence = sentence;
    }

    private void trimStart() {
        while(sentence.startsWith(" ")){
            this.start++;
            sentence = sentence.substring(1);
        }
    }

    @Override
    public String toString() {
        return sentence;
    }

    public int start(){
        return start;
    }

    public int end(){
        return end;
    }

    public int syllableCount(){
        return sc.count(sentence);
    }
}

File: app/src/main/java/com/wbrawner/simplemarkdown/model/Sentence.kt
package com.wbrawner.simplemarkdown.model

import eu.crydee.syllablecounter.SyllableCounter

class Sentence(content: String, private var start: Int = 0, private val end: Int = 0) {

    private var sentence = content.substring(start, end)
    private val sc = SyllableCounter()

    init {
        trimStart()
    }

    private fun trimStart() {
        while (sentence.startsWith(" ")) {
            this.start++
            sentence = sentence.substring(1)
        }
    }

    override fun toString(): String {
        return sentence
    }

    fun start(): Int {
        return start
    }

    fun end(): Int {
        return end
    }

    fun syllableCount(): Int {
        return sc.count(sentence)
    }
}
